File: packages/engine/src/cpus/6502/instructions/sei.js
import { CycleType } from '..'

/**
 * Set Interrupt Disabled
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       SEI           $78  1   2
 */
export function* sei() {
  this.interruptDisabled = true
  yield CycleType.GET
}

File: packages/engine/src/cpus/6502/instructions/sei.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Set Interrupt Disabled
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       SEI           $78  1   2

pub fn sei(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.i = true;
}
